server/mailet/dkim/src/main/java/org/apache/james/jdkim/mailets/MimeMessageHeaders.java

```diff
@@ -49,14 +51,16 @@ public MimeMessageHeaders(MimeMessage message) throws MessagingException {
             }
             String headerName = head.substring(0, p).trim();
             String headerNameLC = headerName.toLowerCase();
-            fields.add(headerName);
-            List<String> strings = headers.get(headerNameLC);
+            fieldsBuilder.add(headerName);
+            List<String> strings = headersBuilding.get(headerNameLC);
             if (strings == null) {
                 strings = new LinkedList<>();
-                headers.put(headerNameLC, strings);
+                headersBuilding.put(headerNameLC, strings);
             }
             strings.add(head);
         }
+        headers = ImmutableMap.copyOf(headersBuilding);
```

Comment:
I would prefer the values of the header to be transformed to an immutable list instead of the mutable LinkedList too.

Reply:
I'm not sure to understand.

As you walk through the map during building you can't have an ImmutableMap.Builder :/

Reply:
:+1: